orc8r/cloud/go/services/state/indexer/reindex/reindexer_singleton_test.go

```diff
@@ -123,27 +124,53 @@ func TestSingletonRun(t *testing.T) {
 	fail3.AssertExpectations(t)
 	require.Equal(t, 2, reindexSuccessNum)
 	require.Equal(t, 5, reindexDoneNum)
+
+	// Test that a network/hardware pair that has been added after Run
+	// will have its states reindexed as well
+
+	ctx, _ := context.WithCancel(context.Background())
+	go reindexer.Run(ctx)
+
```

Comment:
Unless I'm misunderstanding: we shouldn't need to re-run the reindexer, because it's already running

Reply:
I cancelled it in L119 because of the purposefully failing indexers we were running

Reply:
Ah I see. Okay so we want to find a way to test this without canceling+restarting the reindexer -- to ensure that it gracefully handles newly-reported state without requiring a restart

We can either remove the `cancel()`, or move these "more state" checks before checking the failed indexers

Reply:
Ok sounds good, I'll try moving before the failing indexers